Drive/Features/TrackDrive/SlidingCardManager.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SlidingCardManagerViewController.swift
+//  Drive
+//
+//  Created by Amanuel Ketebo on 11/6/20.
+//  Copyright © 2020 com.amanjosh. All rights reserved.
+//
+
+import UIKit
+
+protocol SlidingCardViewController: UIViewController {
+    var fullHeight: CGFloat { get }
+    var partialHeight: CGFloat { get }
+    var almostHiddenHeight: CGFloat { get }
+}
+
+class SlidingCardManager {
+    private var slidingViewController: SlidingCardViewController
+    private var containerViewController: UIViewController
+    private var currentStartingOrigin = CGPoint.zero
+
+    init(slidingViewController: SlidingCardViewController,
+         containerViewController: UIViewController) {
+        self.slidingViewController = slidingViewController
+        self.containerViewController = containerViewController
+    }
+
+    func setUp() {
+        containerViewController.view.addSubview(slidingViewController.view)
+        containerViewController.addChild(slidingViewController)
+
+        // Set up view
+        slidingViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            slidingViewController.view.leftAnchor.constraint(equalTo: containerViewController.view.leftAnchor),
+            slidingViewController.view.rightAnchor.constraint(equalTo: containerViewController.view.rightAnchor),
+            containerViewController.view.bottomAnchor.constraint(equalTo: slidingViewController.view.topAnchor,
+                                                                 constant: slidingViewController.partialHeight),
+            slidingViewController.view.heightAnchor.constraint(equalToConstant: slidingViewController.fullHeight),
+            slidingViewController.view.heightAnchor.constraint(equalToConstant: slidingViewController.fullHeight)
+        ])
+
+        // Set up sliding
+        let gestureRecognizer = UIPanGestureRecognizer(target: self,
+                                                       action: #selector(handlePanGesture(_:)))
+        self.slidingViewController.view.addGestureRecognizer(gestureRecognizer)
+    }
+
+    @objc private func handlePanGesture(_ gestureRecognizer: UIPanGestureRecognizer) {
```

Comment:
This function get's called as you're sliding?

Reply:
Yup 👍🏽